From 51bf6482b663ab49232cb14df6ac6ef113917078 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 26 May 2010 12:40:01 +0800
Subject: [PATCH] ENGR00123841 v4l2 output: crop issue for ic bypass mode

If set crop for input image, and ic bypass mode enabled (after crop,
input and output video have the same size), video will play uncorrect if
input is non-interleaving format. Need set u/v offset for this case.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/media/video/mxc/output/mxc_v4l2_output.c |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/mxc/output/mxc_v4l2_output.c b/drivers/media/video/mxc/output/mxc_v4l2_output.c
index 66a899f..49dcaee 100644
--- a/drivers/media/video/mxc/output/mxc_v4l2_output.c
+++ b/drivers/media/video/mxc/output/mxc_v4l2_output.c
@@ -1579,6 +1579,18 @@ static int mxc_v4l2out_streamon(vout_data * vout)
 			ipu_update_channel_buffer(vout->display_ch,
 				IPU_INPUT_BUFFER,
 				1, vout->v4l2_bufs[vout->ipu_buf[1]].m.offset);
+			if (vout->offset.u_offset || vout->offset.v_offset)
+				/* only update u/v offset */
+				ipu_update_channel_offset(vout->display_ch,
+						IPU_INPUT_BUFFER,
+						vout->v2f.fmt.pix.pixelformat,
+						vout->v2f.fmt.pix.width,
+						vout->v2f.fmt.pix.height,
+						vout->bytesperline,
+						vout->offset.u_offset,
+						vout->offset.v_offset,
+						0,
+						0);
 			ipu_select_buffer(vout->display_ch, IPU_INPUT_BUFFER, 0);
 			ipu_select_buffer(vout->display_ch, IPU_INPUT_BUFFER, 1);
 			queue_work(vout->v4l_wq, &vout->timer_work);
-- 
1.5.4.4

